examples/cifar/main.py

```diff
@@ -63,10 +111,41 @@ def input_shape(self):
     @property
     def output_shape(self):
         return [-1, 10]
+    
+    class Bottleneck(nn.Module):
+        def __init__(self, last_planes, in_planes, out_planes, dense_depth, stride, first_layer):
+            super(CIFAR.Bottleneck, self).__init__()
+            self.out_planes = out_planes
+            self.dense_depth = dense_depth
+
+            self.conv1 = nn.Conv2d(last_planes, in_planes, kernel_size=1, bias=False)
+            self.bn1 = nn.BatchNorm2d(in_planes)
+            self.conv2 = nn.Conv2d(in_planes, in_planes, kernel_size=3, stride=stride, padding=1, groups=32, bias=False)
+            self.bn2 = nn.BatchNorm2d(in_planes)
+            self.conv3 = nn.Conv2d(in_planes, out_planes+dense_depth, kernel_size=1, bias=False)
+            self.bn3 = nn.BatchNorm2d(out_planes + dense_depth)
+
+            self.shortcut = nn.Sequential()
+            if first_layer:
+                self.shortcut = nn.Sequential(
+                    nn.Conv2d(last_planes, out_planes + dense_depth, kernel_size=1, stride=stride, bias=False),
+                    nn.BatchNorm2d(out_planes + dense_depth)
```

Comment:
The DPN network is built by stacking multiple modualized mirco-blocks, the structure of each micro-block is designed with a bottleneck style, which starts
with a 1 × 1 convolutional layer followed by a 3 × 3 convolutional layer, and ends with a 1 × 1 convolutional layer. The output of the last 1 × 1 convolutional layer is split into two parts: the first
part is element-wisely added to the residual path, and the second part is concatenated with the densely
connected path.